src/query/storages/util/src/cached_reader.rs

```diff
@@ -78,19 +78,23 @@ where L: Loader<T> + HasTenantLabel
                     cache_hit: false,
                     read_bytes: 0,
                 };
-                let cache = &mut cache.write().await;
-                match cache.get(location.as_ref()) {
-                    Some(item) => {
+
+                {
+                    let cache = &mut cache.write().await;
```

Comment:
Let's use synchronous locks, which will let the compiler check lock holdings across async. Perhaps #7857 already contains these changes.

Reply:
thanks bro! I am verifying the performance improvement of PR #7857 in EC2 + S3 setup. I'd like to drop this PR if test goes well ;)

Reply:
@zhang2014  performance of PR #7857 is great for the scenario given is this PR's summary:

baseline : cold start, hyperfine, warmup 3, no table cache
~~~
Benchmark 1: mysql -uroot -P3307 -h 127.0.0.1 < explain.sql
  Time (mean ± σ):     268.6 ms ±  28.1 ms    [User: 3.0 ms, System: 1.6 ms]
  Range (min … max):   235.1 ms … 344.8 ms    30 runs
~~~  

cold start, hyperfine, warmup 3 , with table cache enabled

~~~
Benchmark 1: mysql -uroot -P3307 -h 127.0.0.1 < explain.sql
  Time (mean ± σ):      26.0 ms ±   0.5 ms    [User: 2.9 ms, System: 1.5 ms]
  Range (min … max):    25.2 ms …  28.0 ms    30 runs
~~~